### PR TITLE
fix: duplicate revision decryption

### PR DESCRIPTION
**Pull request by mobitar**
- Fixes issue where revisions of an item that is created from a duplication event couldn't be decrypted

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -82,24 +82,25 @@ export { SNPrivilegesService } from './services/privileges_service';
...
+  isNullOrUndefined,
   jsonParseEmbeddedKeys,
+  omitUndefinedCopy,
+  removeFromArray,
+  removeFromIndex,
+  subtractFromArray,
   topLevelCompare,
-  dateSorted
+  truncateHexString,
+  uniqCombineObjArrays,
```
Comment (mobitar) [inline lib/index.ts:103]: Just sorted this alphabetically..

File: lib/protocol/operator/004/operator_004.ts [hunk 2]
```diff
@@ -196,13 +196,26 @@ export class SNProtocolOperator004 extends SNProtocolOperator003 {
   private async authenticatedDataToString(
     attachedData: ItemAuthenticatedData
   ) {
-    return this.crypto.base64Encode(JSON.stringify(sortedCopy(attachedData)));
+    return this.crypto.base64Encode(
+      JSON.stringify(
+        sortedCopy(
+          omitUndefinedCopy(
```
Comment (mobitar) [inline lib/protocol/operator/004/operator_004.ts:202]: Technically JSON.stringify will already omit any keys for which the value is undefined. But I didn't want to count on this being an implementation detail of web browsers. Perhaps React Native does it differently. So this is why we omit undefined keys explicitly.

**Review by johnny243 — CHANGES_REQUESTED** [review]
Tests for remote history seems to be failing

**Comment by mobitar** [comment]
@johnny243 you'd need to run tests in using development environment so that jobs can run, in order for revision tests to succeed.

**Pull request merged**